packages/typescript-estree/src/create-program/useProvidedPrograms.ts

```diff
@@ -12,7 +12,7 @@ import {
 const log = debug('typescript-eslint:typescript-estree:useProvidedProgram');
 
 function useProvidedPrograms(
-  programInstances: ts.Program[],
+  programInstances: ts.Program[] | Iterable<ts.Program>,
```

Comment:
nit - `ts.Program[]` is assignable to `Iterable<ts.Program>`
```suggestion
  programInstances: Iterable<ts.Program>,
```